pkg/logging/logger.go

```diff
@@ -1,13 +1,25 @@
 package logging
 
 import (
+	"flag"
 	"fmt"
 	"os"
 
 	"go.uber.org/zap"
 	"go.uber.org/zap/zapcore"
 )
 
+var (
+	replaceGlobals bool
+	encoding       string
+)
+
+func init() {
```

Comment:
I would avoid using `init` in packages. Can you please move the flags to main and add the encoding as a param to  `NewLogger`, like i did for `logLevel`.

Reply:
@stefanprodan  done. PTAL